components/renku_data_services/notebooks/config/dynamic.py

```diff
@@ -389,18 +406,46 @@ def from_env(cls) -> Self:
             tolerations=yaml.safe_load(StringIO(os.environ.get("", "[]"))),
         )
 
+    @classmethod
+    def _for_testing(cls) -> Self:
+        return cls(
+            culling=_SessionCullingConfig.from_env(),
+            git_proxy=_GitProxyConfig(renku_client_secret="not-defined"),  # nosec B106
+            git_rpc_server=_GitRpcServerConfig.from_env(),
+            git_clone=_GitCloneConfig.from_env(),
+            ingress=_SessionIngress(host="localhost"),
+            ca_certs=_CustomCaCertsConfig.from_env(),
+            oidc=_SessionOidcConfig(
+                client_id="not-defined",
+                client_secret="not-defined",  # nosec B106
+                token_url="http://not.defined",
+                auth_url="http://not.defined",
+                issuer_url="http://not.defined",
+            ),
+            storage=_SessionStorageConfig.from_env(),
+            containers=_SessionContainers.from_env(),
+            ssh=_SessionSshConfig.from_env(),
+            default_image=os.environ.get("", "renku/singleuser:latest"),
+            enforce_cpu_limits=CPUEnforcement(os.environ.get("", "off")),
+            termination_warning_duration_seconds=_parse_value_as_int(os.environ.get("", 12 * 60 * 60)),
+            image_default_workdir="/home/jovyan",
+            node_selector=yaml.safe_load(StringIO(os.environ.get("", "{}"))),
+            affinity=yaml.safe_load(StringIO(os.environ.get("", "{}"))),
+            tolerations=yaml.safe_load(StringIO(os.environ.get("", "[]"))),
+        )
+
 
 @dataclass
 class _K8sConfig:
     """Defines the k8s client and namespace."""
 
-    renku_namespace: str
+    renku_namespace: str = "default"
     sessions_namespace: Optional[str] = None
 
     @classmethod
     def from_env(cls) -> Self:
         return cls(
-            renku_namespace=os.environ["KUBERNETES_NAMESPACE"],
+            renku_namespace=os.environ.get("KUBERNETES_NAMESPACE", "default"),
```

Comment:
yes I know how environ works.
I don't know what K8s does with a namespace of `default`. Because the `default` namespace in any of our clusters is empty and not used, so `default` is a bad default for us. 
I was just wondering if `default` is some magic value that automatically picks e.g. `renku` in our cluster. I don't think that's the case, but then I don't really know why we would use `default` as the default, as that won't work in any of our deployments and erroring here might be a better choice.

Reply:
Somehow I knew I was answering the wrong question, sorry.

I think there are two aspects here:
- The namespace where Renku itself is deployed
- The namespace where the sessions are created

Are they always the same ?

Reply:
they can be different but in practice they are not, we don't use that setting anywhere to my knowledge.

Reply:
So this is leftover from a feature that we partly or fully implemented on notebooks. Where we wanted Renku services to run in one namespace and sessions to run in a different namespaces. But we never tested it properly and we never deployed. So we can fully remove `sessions_namespace`. I will do that.